core/connector/src/main/java/org/exoplatform/ecm/connector/dlp/FileDlpConnector.java

```diff
@@ -152,4 +183,108 @@ private void saveDlpPositiveItem(Node node) throws RepositoryException {
+    Node sameNameNode = ((Node) session.getItem(restorePath));
+    Node parent = sameNameNode.getParent();
+    String name = sameNameNode.getName();
+    NodeIterator nodeIter = parent.getNodes(name);
+    while (nodeIter.hasNext()) {
+      Node node = nodeIter.nextNode();
+      if (node.isNodeType(EXO_RESTORE_LOCATION))
+        node.removeMixin(EXO_RESTORE_LOCATION);
+    }
+  }
+
+  private void addRestorePathInfo(String nodeName, String restorePath, String nodeWorkspace) throws Exception {
+    NodeIterator nodes = this.getSecurityHomeNode().getNodes(nodeName);
+    Node node = null;
+    while (nodes.hasNext()) {
+      Node currentNode = nodes.nextNode();
+      if (node == null) {
+        node = currentNode;
+      } else {
+        if (node.getIndex() < currentNode.getIndex()) {
+          node = currentNode;
+        }
+      }
+    }
+    if (node != null) {
+      node.addMixin(EXO_RESTORE_LOCATION);
+      node.setProperty(RESTORE_PATH, restorePath);
+      node.setProperty(RESTORE_WORKSPACE, nodeWorkspace);
+      node.save();
+    }
+  }
+
+  private Node getSecurityHomeNode() {
+    try {
+      Session session = WCMCoreUtils.getSystemSessionProvider()
+                                    .getSession(COLLABORATION_WS,
+                                                repositoryService.getCurrentRepository());
+      return (Node) session.getItem("/" + DLP_SECURITY_FOLDER);
+    } catch (Exception e) {
+      return null;
+    }
+
+  }
+
+  private String fixRestorePath(String path) {
+    int leftBracket = path.lastIndexOf('[');
+    int rightBracket = path.lastIndexOf(']');
+    if (leftBracket == -1 || rightBracket == -1 ||
+        (leftBracket >= rightBracket)) return path;
+
+    try {
+      Integer.parseInt(path.substring(leftBracket+1, rightBracket));
+    } catch (Exception ex) {
+      return path;
+    }
+    return path.substring(0, leftBracket);
+  }
+
+  private void restoreFromSecurity(String SecurityNodePath,
+                                SessionProvider sessionProvider) throws Exception {
+
+    LinkManager linkManager = CommonsUtils.getService(LinkManager.class);
+    Node securityHomeNode = this.getSecurityHomeNode();
+    Session securityNodeSession = securityHomeNode.getSession();
+    Node securityNode = (Node)securityNodeSession.getItem(SecurityNodePath);
+    String securityWorkspace = securityNodeSession.getWorkspace().getName();
+    String restoreWorkspace = securityNode.getProperty(RESTORE_WORKSPACE).getString();
+    String restorePath = securityNode.getProperty(RESTORE_PATH).getString();
+
+    ManageableRepository manageableRepository = repositoryService.getCurrentRepository();
+    Session restoreSession = sessionProvider.getSession(restoreWorkspace,  manageableRepository);
+
+    if (restoreWorkspace.equals(securityWorkspace)) {
+      securityNodeSession.getWorkspace().move(SecurityNodePath, restorePath);
+    } else {
+      //clone node
+      restoreSession.getWorkspace().clone(
```

Comment:
Does this case exists ?
I think the FileDlpAction should be configured only for collaboration workspace, so we cannot have a file which comes from another workspace